azurerm/internal/services/applicationinsights/resource_arm_application_insights.go

```diff
@@ -64,6 +64,23 @@ func resourceArmApplicationInsights() *schema.Resource {
 				}, true),
 			},
 
+			"retention_in_days": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  90,
```

Comment:
Unfortunately, this is a breaking change since the API doesn't provide `retention_in_days` by default. This means that users migrating to this update will see a change like `retention_in_days   = 0 -> 90`.  We can't have this so this default should be removed.

Reply:
If the API returns a default we can probably just use computed instead

Reply:
API returns "nil". So I removed the default value.